jwt/api_jws.py

```diff
@@ -127,15 +127,14 @@ def encode(
 
         return encoded_string.decode("utf-8")
 
-    def decode(
+    def decode_complete(
         self,
         jwt: str,
         key: str = "",
         algorithms: List[str] = None,
         options: Dict = None,
-        complete: bool = False,
         **kwargs,
-    ):
+    ) -> Dict[str, Any]:
```

Comment:
This is a perfectly fine solution, but if you wanted to maintain backwards-compatibility, this could remain the same, but use `@overload`.
It would look something like:
```
@overload
def decode(self, jwt: str, key: str = ..., algorithms: Optional[List[str]] = ..., options: Optional[Dict] = ..., complete: Literal[True], **kwargs: Any) -> Dict[str, Any]: ...
@overload
def decode(self, jwt: str, key: str = ..., algorithms: Optional[List[str]] = ..., options: Optional[Dict] = ..., complete: Literal[False] = ..., **kwargs: Any) -> str: ...
def decode(self, jwt: str, key: str = "", algorithms: Optional[List[str]] = None, options: Optional[Dict] = None, complete: bool = False, **kwargs) -> Union[str, Dict[str, Any]]:
```

Reply:
We're prepping for version 2.0, so this is the chance to break backward compatibility if we must. IMO, we should aim for the desired interface rather than force support for something we can drop.

Reply:
I played around with the `overload` idea a bit. Unfortunately, `Literal` was only introduced in Python 3.8.

Reply:
Yes, that's right. Not sure you can do the overload without it.
If you wanted to go that route, you could import from `typing_extensions` for backwards compatibility (adding that to the dependencies). Alternatively, only do the overload in Python 3.8+ (`if sys.version_info >= (3, 8)`, and users of older versions will have to put up with the `Union` (though I suspect most Mypy users are already using 3.8+).

Reply:
Your current proposal is probably the easier option though.